pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -165,9 +165,14 @@ func (p *PodAutoscalerInternal) UpdateFromSettings(podAutoscalerSpec *datadoghq.
 	p.horizontalEventsRetention = getHorizontalEventsRetention(podAutoscalerSpec.Policy, longestScalingRulePeriodAllowed)
 }
 
-// UpdateFromValues updates the PodAutoscalerInternal scaling values
-func (p *PodAutoscalerInternal) UpdateFromValues(scalingValues ScalingValues) {
-	p.scalingValues = scalingValues
+// UpdateFromValues updates the PodAutoscalerInternal scaling values based on the desired source of recommendations
+func (p *PodAutoscalerInternal) UpdateFromValues(source datadoghq.DatadogPodAutoscalerValueSource) {
```

Comment:
I think keeping the name is a bit misleading, we're not really updating from values but more making values primary/actual (or selecting/switching, whatever name you prefer)